pkg/controller/memberstatus/memberstatus_controller.go

```diff
@@ -0,0 +1,305 @@
+package memberstatus
+
+import (
+	"context"
+	"fmt"
+	"os"
+	"time"
+
+	toolchainv1alpha1 "github.com/codeready-toolchain/api/pkg/apis/toolchain/v1alpha1"
+	"github.com/codeready-toolchain/member-operator/pkg/configuration"
+	"github.com/codeready-toolchain/member-operator/version"
+	"github.com/codeready-toolchain/toolchain-common/pkg/cluster"
+
+	"github.com/go-logr/logr"
+	errs "github.com/pkg/errors"
+
+	appsv1 "k8s.io/api/apps/v1"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/types"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/controller"
+	"sigs.k8s.io/controller-runtime/pkg/handler"
+	logf "sigs.k8s.io/controller-runtime/pkg/log"
+	"sigs.k8s.io/controller-runtime/pkg/manager"
+	"sigs.k8s.io/controller-runtime/pkg/predicate"
+	"sigs.k8s.io/controller-runtime/pkg/reconcile"
+	"sigs.k8s.io/controller-runtime/pkg/source"
+	kubefed_v1beta1 "sigs.k8s.io/kubefed/pkg/apis/core/v1beta1"
+	"sigs.k8s.io/kubefed/pkg/controller/util"
+)
+
+var log = logf.Log.WithName("controller_memberstatus")
+
+// general memberstatus constants
+const (
+	OperatorNameVar = "OPERATOR_NAME"
+
+	defaultMemberStatusName = "toolchain-member-status"
+	defaultRequeueTime      = time.Second * 5
+)
+
+// custom component condition error messages & reasons
+const (
+	// deployment not found
+	reasonNoDeployment         = "DeploymentNotFound"
+	errMsgNoMemberOperatorName = "unable to look up the member operator name, the environment variable OPERATOR_NAME is not set"
+	errMsgCannotGetDeployment  = "unable to get the member operator deployment"
+
+	// kubefed not found
+	reasonHostConnectionNotFound = "KubefedNotFound"
```

Comment:
Shouldn't be the reasons specified in api repo? It can be then also reused in host-operator. But only the reasons - the errors should stay here.

Reply:
I thought about that but by the time I did I already made the updates to the api repo and didn't want to go through that hell again just for these strings so I'll just do it when I update the api for the status changes on the host side. Added it to the completion steps for the host operator issue.

Reply:
If you want to move the constants to api repo later then I'm fine, but you will need to change it in the member-operator repo (and go through that hell again) anyway 
But it's up to you ;-)